app/scripts/lib/rpc-method-middleware/handlers/send-metadata.ts

```diff
@@ -0,0 +1,81 @@
+import { ethErrors } from 'eth-rpc-errors';
+import {
+  JsonRpcParams,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from '@metamask/utils';
+import {
+  JsonRpcEngineEndCallback,
+  JsonRpcEngineNextCallback,
+} from '@metamask/json-rpc-engine';
+import { SubjectType } from '@metamask/permission-controller';
+import { MESSAGE_TYPE } from '../../../../../shared/constants/app';
+import { AddSubjectMetadata, HandlerWrapper } from './handlers-helper';
+
+type SendMetadataOptions = {
+  addSubjectMetadata: AddSubjectMetadata;
+  subjectType: SubjectType;
+};
+
+type SendMetadataConstraints<Params extends JsonRpcParams = JsonRpcParams> = {
```

Comment:
Nit: since the entire type object is the type constraint.

```suggestion
type SendMetadataConstraint<Params extends JsonRpcParams = JsonRpcParams> = {
```